PROTO_tests/tests/indexing_test.py

```diff
@@ -103,3 +103,8 @@ def test_bigint_indexing_preserves_max_bits(self):
         a = ak.arange(2**200 - 1, 2**200 + 11, max_bits=max_bits)
         assert max_bits == a[ak.arange(10)].max_bits
         assert max_bits == a[:].max_bits
+
+    def test_handling_bigint_max_bits(self):
+        a = ak.arange(2**200 - 1, 2**200 + 11)
+        a[:] = ak.array(a, dtype=ak.bigint, max_bits=3)
+        assert [7, 0, 1, 2, 3, 4, 5, 6, 7, 0, 1, 2] == a.to_list()
```

Comment:
this still isn't testing what u think it is, this test will still work on master. I ran the below python block on master, hopefully it'll make it a bit easier to see what the problem is. If not, we can jump on a call and talk through it
```python
# this is what you test is doing, as you can see it works on master without your fix
>>> a = ak.arange(2**200 - 1, 2**200 + 11)
>>> a[:] = ak.array(a, dtype=ak.bigint, max_bits=3)
>>> a
array([7 0 1 2 3 4 5 6 7 0 1 2])

# this is what's broken on master
>>> a = ak.arange(2**200 - 1, 2**200 + 11, max_bits=3)
>>> a[:] = ak.arange(2**200 - 1, 2**200 + 11)
>>> a
array([0 1 2 0 1 2 0 1 2 0 1 2])
```
A good way to avoid this in the future is to write your tests first for bug fixes. By verifying that a test was failing and is no longer, you can have more confidence you actually fixed the problem